Codigo/CustomScenarios.bas

```diff
@@ -85,17 +116,23 @@ End Function
 Public Sub RemoveUpdateScenario(ByRef Index As Integer)
     Debug.Assert Index < UBound(ScenarioUpdateList)
     Set ScenarioUpdateList(Index) = Nothing
+    Dim i As Integer
+    For i = 0 To ActiveUpdateSlots.currentIndex - 1
+        If ActiveUpdateSlots.IndexInfo(i) = Index Then
+            ActiveUpdateSlots.IndexInfo(i) = ActiveUpdateSlots.IndexInfo(ActiveUpdateSlots.currentIndex - 1)
+            ActiveUpdateSlots.currentIndex = ActiveUpdateSlots.currentIndex - 1
+            AvailableUpdateSlots.currentIndex = AvailableUpdateSlots.currentIndex + 1
+            AvailableUpdateSlots.IndexInfo(AvailableUpdateSlots.currentIndex) = Index
+        End If
+    Next
 End Sub
 
 Public Sub UpdateAll()
 On Error GoTo UpdateAll_Err:
-    If Not IsArrayInitialized(ScenarioUpdateList) Then
-        Exit Sub
-    End If
     Dim i As Integer
-    For i = 0 To UBound(ScenarioUpdateList)
-        If Not ScenarioUpdateList(i) Is Nothing Then
-            Call ScenarioUpdateList(i).Update
+    For i = 0 To ActiveUpdateSlots.currentIndex - 1
```

Comment:
As in here,
`For i = 0 To UBound(ScenarioUpdateList)` is much nicer than `For i = 0 To ActiveUpdateSlots.currentIndex - 1`.

Why the extra complexity?